code/Bridge/MonologCollector.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace LeKoala\DebugBar\Bridge;
+
+use DebugBar\DataCollector\DataCollectorInterface;
+use DebugBar\DataCollector\MessagesAggregateInterface;
+use DebugBar\DataCollector\Renderable;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Logger;
+use Monolog\LogRecord;
+
+/**
+ * A monolog handler as well as a data collector
+ * Based on DebugBar\Bridge\MonologCollector
+ * <code>
+ * $debugbar->addCollector(new MonologCollector($logger));
+ * </code>
+ */
+class MonologCollector extends AbstractProcessingHandler implements DataCollectorInterface, Renderable, MessagesAggregateInterface
```

Comment:
Why are we copy pasting this in from https://github.com/maximebf/php-debugbar/blob/master/src/DebugBar/Bridge/MonologCollector.php?

`"maximebf/debugbar": "^1.18"`, which should be PHP 8.1 compatible, is already included in composer.json?

Reply:
It looks like this method declaration https://github.com/maximebf/php-debugbar/blob/v1.18.1/src/DebugBar/Bridge/MonologCollector.php#L62 is not compatible with `Monolog\Handler\AbstractProcessingHandler::write(Monolog\LogRecord $record): void`. It still fail our tests https://github.com/lekoala/silverstripe-debugbar/actions/runs/3934516960/jobs/6729336460#step:12:62.

This issue was already created https://github.com/maximebf/php-debugbar/issues/514

Reply:
OK, probably need to do this then to workout this issue

Could you please put a class docblock explaining the reason why this class exists, and mention that it should be deprecated and stop being used once the maximebf/php-debugbar works with monolog/monolog:^3

Reply:
DONE